src/PostgrestFilterBuilder.ts

```diff
@@ -269,9 +273,13 @@ export default class PostgrestFilterBuilder<
    */
   in<ColumnName extends string>(
     column: ColumnName,
-    values: ResolveFilterValue<Schema, Row, ColumnName> extends never
-      ? unknown[]
-      : ReadonlyArray<ResolveFilterValue<Schema, Row, ColumnName>>
+    values: ReadonlyArray<
+      ResolveFilterValue<Schema, Row, ColumnName> extends infer ResolvedFilterValue
+        ? ResolvedFilterValue extends never
+          ? unknown[]
+          : ResolvedFilterValue
+        : never
+    >
```

Comment:
Here we need the `infer` within the `ReadonlyArray` to have the proper resulting type, otherwise we'll have an invalid union.